cmd/ipfs-cluster-service/main.go

```diff
@@ -632,3 +641,40 @@ func getStateManager() cmdutils.StateManager {
 	checkErr("creating state manager", err)
 	return mgr
 }
+
+func assignRandomPorts(multiAddrs []*ma.Multiaddr) {
+	for _, m := range multiAddrs {
+		var withRandom ma.Multiaddr
```

Comment:
would it simplify to keep `components []ma.Multiaddr` and call `multiaddr.Join(components)` at the end?